synapse/storage/databases/main/event_federation.py

```diff
@@ -966,30 +988,32 @@ async def get_forward_extremities_for_room_at_stream_ordering(
         """
         # We want to make the cache more effective, so we clamp to the last
         # change before the given ordering.
-        last_change = self._events_stream_cache.get_max_pos_of_last_change(room_id)
+        last_change = self._events_stream_cache.get_max_pos_of_last_change(room_id)  # type: ignore[attr-defined]
```

Comment:
`_events_stream_cache` is defined on StreamWorkerStore. Could try to make this class inherit from StreamWorkerStore if you were feeling brave.